var/spack/repos/builtin/packages/3proxy/package.py

```diff
@@ -14,6 +14,8 @@ class _3proxy(MakefilePackage):
     homepage = "https://3proxy.org"
     url = "https://github.com/z3APA3A/3proxy/archive/0.8.13.tar.gz"
 
+    license("Apache-2.0 GPL-2.0-or-later LGPL-2.1-or-later")
```

Comment:
```suggestion
    license("Apache-2.0", "GPL-2.0-or-later", "LGPL-2.1-or-later")
```

I may have missed this in the initial PR, but I thought multiple licenses would be declared like multiple maintainers?

Reply:
They're supposed to be valid [SPDX expressions](https://spdx.github.io/spdx-spec/v2.3/SPDX-license-expressions/). The `license` directive only takes in a single license expression and an optional when spec, and they all have to match mutually exclusive specs. There are a couple expressions that aren't valid at the moment (including this one) as it should be using the `OR` operator. To properly validate these, we need a compliant expression parser. I'm planning on implementing one in the near future, but not sure when I'll be able to get to it.

The plan (on my end) was as follows:
1. Get initial automatically collected license information in (this PR).
2. Add in a SPDX expression parser.
3. Enable the use of the expression parser for validating package license expressions/fix existing invalid expressions.
4. More work to better integrate licenses into Spack.

Landing this in some form (or I guess I could just pull the commit into my fork...) unblocks me on a couple downstream tasks and would also enable large scale testing of a couple other patches I have planned, such as adding license information to https://packages.spack.io.

Reply:
Most of these issues should be fixed now with my latest update, more details below.